pkg/cluster/clustermsi.go

```diff
@@ -0,0 +1,172 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/msi-dataplane/pkg/dataplane"
+	"github.com/Azure/msi-dataplane/pkg/store"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armmsi"
+)
+
+const (
+	msiCertValidityDays = 90
+)
+
+// ensureClusterMsiCertificate leverages the MSI dataplane module to create a new MSI
```

Comment:
Nit: `module to fetch the backing certificate..`

Reply:
I pushed an update here. Let me know if I didn't correctly capture your intended meaning!